javascript/javascript1/week3/noteTakingApp.js

```diff
@@ -0,0 +1,50 @@
+//Smart-ease - Back to the basics!
+
+//save a note
+
+const notes = [];
+
+function saveNote(content, id) {
+  // write some code here
+  
+  notes.push({ content: content , id: id  });
+  
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("cook food", 3);
+saveNote("clean the house", 4);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+
+//get a note
+
+function getNote(id) {
+    if (id >= notes.length) {
+        console.log("The required ID does not exist");
+        return
+    }
```

Comment:
I ran your code and got `The required ID does not exist` but note with ID = 4 does exist. The problem is this if statement. It is not relevant because `id` is not the array index. The array looks like this.
```
[
 0: { content: 'Pick up groceries', id: 1 },
 1: { content: 'Do laundry', id: 2 },
 2: { content: 'cook food', id: 3 },
 3: { content: 'clean the house', id: 4 }
]
```
The `id` you are using goes from 1 to 4, so the note with ID = 4 does exist. Notice that this is different to the numbers 0 to 3, which are the array indexes.